components/raftstore/src/store/entry_storage.rs

```diff
@@ -248,8 +253,12 @@ impl EntryCache {
                 }
                 break;
             }
-            let (_, dangle_size) = cached_entries.take_entries();
-            mem_size_change -= dangle_size as i64;
+            // keep the applying entries if not all of them are persisted.
+            if cached_entries.range.end <= idx {
```

Comment:
How about keeping these entries only when #16834 is enabled?

Reply:
If #16834 is disabled, then the range end is impossible to be bigger than persisted index, this is also the reason that the origin logic does not need to check the range end.